### DISCUSSION TITLE
ProcessOutboxJob

### DISCUSSION THREAD

**Discussion by didiercauvin** [kgrzybek/modular-monolith-with-ddd#224] [Q&A] [answered]
Hi,

my question is about ProcessOutboxJob. As this job is planned to be played every two seconds, what happened if an event handler takes more than 2 seconds to be executed? The next call to ProcessOutboxJob will play the events not already played by the previous one, as they are not marked processed, and the last one will execute the rest of the events in its list, right? So they will be played twice.  Am I right or is there something I've missed?

**Comment by AndreiGanichev** [comment, marked as answer]
Hi, @didiercauvin.
You missed that `DisallowConcurrentExecutionAttribute` is added to [ProcessOutboxJob](https://github.com/kgrzybek/modular-monolith-with-ddd/blob/master/src/Modules/Administration/Infrastructure/Configuration/Processing/Outbox/ProcessOutboxJob.cs) class. According to [documentation](https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/more-about-jobs.html#job-state-and-concurrency) the attribute:
> can be added to the Job class that tells Quartz not to execute multiple instances of a given job definition (that refers to the given job class) concurrently.

**Reply by patxiba** [reply]
What if the API is installed in a cluster with two servers that provide high availability?
